app/controllers/api/v2/job_invocations_controller.rb

```diff
@@ -85,12 +85,32 @@ def output
         end
       end
 
+      api :POST, '/job_invocations/:id/cancel'
+      param :id, :identifier, :required => true
+      param :force, :bool
+      def cancel
+        unless Authorizer.new(User.current, :collection => [@job_invocation.task]).can? :edit_foreman_tasks
+          render :json => { :error => _('The user is not authorized to cancel the job') },
+                 :status => 403
+        end
+
+        if @job_invocation.task.cancellable?
+          result = @job_invocation.cancel(params.fetch(['force'], false))
+          render :json => { :cancelled => result, :id => @job_invocation.id }
+        else
+          render :json => { :message => _('The job could not be cancelled.') },
+                 :status => 422
+        end
+      end
+
       private
 
       def action_permission
         case params[:action]
         when 'output'
           :view
+        when 'cancel'
+          :view
```

Comment:
Should this be `:view`?

Reply:
The user needs to be able to view the job invocation and edit tasks. I can create a dedicated permission for editing job invocations if needed, so far we don't have one.

Reply:
+1 for addiing it.